backend/app/services/research_financials.py

```
@@ -0,0 +1,282 @@
+
+        entry = db.query(AppSetting).filter(AppSetting.key == key).first()
+        if entry is None:
+            return None
+        ts_part, _, json_part = entry.value.partition("|")
+        if not ts_part or not json_part:
+            return None
+        fetched_at = datetime.fromisoformat(ts_part)
+        payload = json.loads(json_part)
+        if not isinstance(payload, dict):
+            return None
+        return payload, fetched_at
```

Ruff (0.15.13)

[warning] Consider moving this statement to an `else` block

(TRY300)